includes/class-wc-payments.php

```diff
@@ -689,4 +695,150 @@ public static function remove_woo_admin_notes() {
+			self::build_and_save_translations( $locale );
+		}
+	}
+
+	/**
+	 * Combine and save translations for a specific locale.
+	 *
+	 * Note that this assumes \WP_Filesystem is already initialized with write access.
+	 *
+	 * @param string $locale Locale being retrieved.
+	 */
+	public static function build_and_save_translations( $locale ) {
+		// So long as this function is called during the 'upgrader_process_complete' action,
+		// the filesystem object should be hooked up.
+		global $wp_filesystem;
+
+		$translations_from_chunks = self::get_translation_chunk_data( $locale );
+
+		if ( empty( $translations_from_chunks ) ) {
+			return;
+		}
+
+		$cache_filename          = self::get_combined_translation_filename( $locale );
+		$chunk_translations_json = wp_json_encode( $translations_from_chunks );
+
+		// Cache combined translations strings to a file.
+		$wp_filesystem->put_contents( WP_LANG_DIR . '/plugins/' . $cache_filename, $chunk_translations_json );
+	}
+
+	/**
+	 * Find and combine translation chunk files.
+	 *
+	 * Only targets files that aren't represented by a registered script (e.g. not passed to wp_register_script()).
+	 *
+	 * @param string $locale Locale being retrieved.
+	 * @return array Combined translation chunk data.
+	 */
+	public static function get_translation_chunk_data( $locale ) {
+		global $wp_filesystem;
+
+		// Grab all JSON files in the current language pack.
+		$json_i18n_filenames       = glob( WP_LANG_DIR . '/plugins/woocommerce-payments-' . $locale . '-*.json' );
+		$combined_translation_data = [];
+
+		if ( false === $json_i18n_filenames ) {
+			return $combined_translation_data;
+		}
+
+		foreach ( $json_i18n_filenames as $json_filename ) {
+			if ( ! $wp_filesystem->is_readable( $json_filename ) ) {
+				continue;
+			}
+
+			$file_contents = $wp_filesystem->get_contents( $json_filename );
+			$chunk_data    = \json_decode( $file_contents, true );
+
+			if ( empty( $chunk_data ) ) {
+				continue;
+			}
+
+			if ( empty( $combined_translation_data ) ) {
+				// Use the first translation file as the base structure.
+				$combined_translation_data = $chunk_data;
+			} else {
+				// Combine all messages from all chunk files.
+				$combined_translation_data['locale_data']['messages'] = array_merge(
+					$combined_translation_data['locale_data']['messages'],
+					$chunk_data['locale_data']['messages']
+				);
+			}
+		}
+
+		return $combined_translation_data;
+	}
+
+	/**
+	 * Load translation strings from language packs for dynamic imports.
+	 *
+	 * @param string $file File location for the script being translated.
+	 * @param string $handle Script handle.
+	 * @param string $domain Text domain.
+	 *
+	 * @return string New file location for the script being translated.
+	 */
+	public static function load_script_translation_file( $file, $handle, $domain ) {
+		// Make sure the main app script is being loaded.
+		if ( false === strpos( $handle, 'WCPAY_' ) ) {
+			return $file;
+		}
```

Comment:
I think a better check here would be `0 !== strpos( $handle, 'WCPAY_' )` to make sure we're checking for the prefix 🙂 . At least I think all our scripts start with `WCPAY_` otherwise that wouldn't really work 😅 

The way this currently works any script with `WCPAY_` anywhere (+ the `woocommerce-payments` domain) would load the translation files 🙂. I don't really think that's a problem, but it doesn't hurt to be a bit more precise here I think ☺️

Reply:
Good catch!